stingray/modeling/posterior.py

```diff
@@ -126,7 +131,12 @@ def logprior(t0, neg=False):
         # correctly!
         for pname in lpost.model.param_names:
             if not lpost.model.fixed[pname]:
-                logp += np.log(priors[pname](t0[ii]))
+                with warnings.catch_warnings(record=True) as out:
+                    logp += np.log(priors[pname](t0[ii]))
+                    if len(out) > 0:
```

Comment:
(BTW: `out` _will_ be still accessible out of the `with` statement. I have a couple more examples of this, which - I agree - seems to clash with the whole point of the `with` statement :) )